test/general/buffer_wrapper.pass.cpp

```diff
@@ -34,7 +34,9 @@ struct test_buffer_wrapper
         EXPECT_TRUE(end - begin == size, "wrong effect of iterator's operator - iterator");
 
         auto buf = begin.get_buffer();
-        T* actual_data = sycl::host_accessor<T, 1, sycl::access_mode::read>(buf).get_pointer();

```

Comment:
1) By following up to the issue #788 where we have  got `'no matching conversion for functional-style cast from 'sycl::buffer<unsigned int, dim, aligned_allocator<unsigned int>>' to 'sycl::host_accessor<const unsigned int, 1, sycl::access_mode::read>'`
It tells that we have a buffer parametrized by `unsigned int` and `sycl::host_accessor` parametrized  by `const unsigned int`. 
2) The proposed changes probably doesn't resolve the issue because the SYCL spec tells " using value_type = // const DataT for read-only accessors, DataT otherwise" ,  so we have got the same error...

3) Furthermore, looking into the SYCL spec I cannot see such constructor with a buffer and access mode, how it is proposed in the changes.

Reply:
About (3) - please see my comments at https://github.com/oneapi-src/oneDPL/pull/793#discussion_r1104332954